ci/Dockerfile-envoy-windows

```diff
@@ -3,9 +3,12 @@ ARG BUILD_TAG=ltsc2019
 
 FROM $BUILD_OS:$BUILD_TAG
 
+ARG BUILD_TYPE=release
+
 RUN mkdir "C:\\Program\ Files\\envoy"
 RUN setx path "%path%;c:\Program Files\envoy"
-ADD ["windows/amd64/envoy.exe", "C:/Program Files/envoy/"]
+ADD ["windows/amd64/envoy.*", "C:/Program Files/envoy/"]
```

Comment:
Only one concern, is this limited to envoy.exe and envoy.pdb? The linker often emits other crazy output such as .map and .lib files, even for .exe files.

Reply:
the `windows/amd64` directory will only contain what we put in the tarball we created here: https://github.com/envoyproxy/envoy/pull/14265/files#diff-c2b36673c33d9d04ccfde19e54e4a5273d69b47523a258f91de9c71e5ef87e61R76

(tarball downloaded here and untarred) https://github.com/envoyproxy/envoy/blob/d1ded6b381ca92cbacb2e0683adf997239b12272/.azure-pipelines/pipelines.yml#L418-L430

Reply:
The build artifact seems to include the tarball and the tree the tarball is created from, is this an issue?

Reply:
Moving this q... comment... etc over to discussions of windows_build_steps.sh